scripts/updateLPsAPR/index.ts

```diff
@@ -0,0 +1,136 @@
+import fs from 'fs'
+import { request, gql } from 'graphql-request'
+import BigNumber from 'bignumber.js'
+import { ChainId } from '@pancakeswap-libs/sdk'
+import chunk from 'lodash/chunk'
+import farmsConfig from '../../src/config/constants/farms'
+
+const BLOCK_SUBGRAPH = 'https://api.thegraph.com/subgraphs/name/pancakeswap/blocks'
+const STREAMING_FAST = 'https://bsc.streamingfast.io/subgraphs/name/pancakeswap/exchange-v2'
+
+const GET_BLOCK = (timestamp: string) => {
+  return gql`query getBlock {
+        blocks(first: 1, where: { timestamp_gt: ${timestamp}, timestamp_lt: ${timestamp + 600} }) {
+          number
+        }
+    }`
+}
+
+const FARMS_AT_BLOCK = (block: number | null, farms: string[]) => {
+  const blockString = block ? `block: {number: ${block}}` : ``
+  const addressesString = `["${farms.join('","')}"]`
+  return `pairs(
+      first: 30,
+      where: { id_in: ${addressesString} }
+      ${blockString}
+    ) {
+      id
+      volumeUSD
+      reserveUSD
+    }`
+}
+
+const FARMS_BULK = (blockWeekAgo: number, farms: string[]) => {
+  return gql`
+      query farmsBulk {
+        now: ${FARMS_AT_BLOCK(null, farms)}
+        oneWeekAgo: ${FARMS_AT_BLOCK(blockWeekAgo, farms)}
+      }
+    `
+}
+
+interface BlockResponse {
+  blocks: {
+    number: number
+  }[]
+}
+
+interface SingleFarmResponse {
+  id: string
+  reserveUSD: string
+  volumeUSD: string
+}
+
+interface FarmsResponse {
+  now: SingleFarmResponse[]
```

Comment:
now?

Reply:
now as in "at latest available block right now"

Reply:
renamed to `farmsAtLatestBlock`